sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -380,29 +404,44 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
      */
     public Mono<MsalToken> authenticateWithUsernamePassword(TokenRequestContext request,
                                                             String username, String password) {
-        return Mono.fromFuture(publicClientApplication.acquireToken(
+        return Mono.fromFuture(() -> getPublicClientApplication(false).acquireToken(
             UserNamePasswordParameters.builder(new HashSet<>(request.getScopes()), username, password.toCharArray())
                 .build()))
-            .map(ar -> new MsalToken(ar, options));
+            .onErrorResume(t -> Mono.error(new ClientAuthenticationException("Failed to acquire token with username "
+                + "and password", null, t))).map(ar -> new MsalToken(ar, options));
     }
 
     /**
      * Asynchronously acquire a token from the currently logged in client.
      *
      * @param request the details of the token request
+     * @param account the account used to login to acquire the last token
      * @return a Publisher that emits an AccessToken
      */
-    public Mono<MsalToken> authenticateWithUserRefreshToken(TokenRequestContext request, MsalToken msalToken) {
+    public Mono<MsalToken> authenticateWithMsalAccount(TokenRequestContext request, IAccount account) {
         SilentParameters parameters;
-        if (msalToken.getAccount() != null) {
-            parameters = SilentParameters.builder(new HashSet<>(request.getScopes()), msalToken.getAccount()).build();
+        SilentParameters forceParameters;
+        if (account != null) {
+            parameters = SilentParameters.builder(new HashSet<>(request.getScopes()), account).build();
```

Comment:
Does account here have to be non-null? If it can be null, could the if/else block be merged to always pass `account`?

Reply:
Unfortunately in the first builder overload the account is required to be non-null.

Reply:
But it does seem like I can add account afterwards

Reply:
Reorganized a little bit to avoid creating 2 parameters upfront